### PR TITLE
[SPARK-38289][SQL] Refactor SQL CLI exit code to make it more clear

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
In this pr we define exit code use in SQL CLI and refactor the code.
After this change can make the exit relate code more clear. 

### Why are the changes needed?
Make code more clear

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not need

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
gentle ping @yaooqinn @wangyum @cloud-fan

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 import java.util.Properties;
 
+import org.apache.spark.util.SparkExitCode;
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java]: > import order

Updated

**Comment by LuciferYang** [comment]
I found that there are many similar cases in Spark code, for example:

https://github.com/apache/spark/blob/c4b013f4129969668323d20e40869411b35ed5e7/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L269-L277

https://github.com/apache/spark/blob/c4b013f4129969668323d20e40869411b35ed5e7/core/src/main/scala/org/apache/spark/api/r/RBackend.scala#L111-L118

and so on. Shall we fix them all in one?

**Comment by AngersZhuuuu** [comment]
> I found that there are many similar cases in Spark code, for example:
> 
> https://github.com/apache/spark/blob/c4b013f4129969668323d20e40869411b35ed5e7/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L269-L277
> 
> https://github.com/apache/spark/blob/c4b013f4129969668323d20e40869411b35ed5e7/core/src/main/scala/org/apache/spark/api/r/RBackend.scala#L111-L118
> 
> and so on. Shall we fix them all in one?

So many, and all of these exit code is `-1`, `1`, `0`, I think it's easy to understand. Not so necessary

**Comment by LuciferYang** [comment]
> So many, and all of these exit code is -1, 1, 0, I think it's easy to understand. Not so necessary

OK

**Comment by AngersZhuuuu** [comment]
gentle ping @HyukjinKwon @cloud-fan @dongjoon-hyun

**Comment by LuciferYang** [comment]
> So many, and all of these exit code is -1, 1, 0, I think it's easy to understand. Not so necessary

I thought again, if -1, 0 and 1 are easy to understand, I think we don't need to name them, just name the states that are not easy to understand is enough

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java [hunk 2]
```diff
@@ -259,7 +260,7 @@ static class HelpOptionExecutor implements ServerOptionsExecutor {
     @Override
     public void execute() {
       new HelpFormatter().printHelp(serverName, options);
-      System.exit(0);
+      System.exit(SparkExitCode.EXIT_SUCCESS());
```
Comment (cloud-fan) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java:263]:
```suggestion
      System.exit(SparkExitCode.EXIT_SUCCESS);
```

**Review by cloud-fan — APPROVED** [review]

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Tests are running in https://github.com/AngersZhuuuu/spark/runs/6765635006

Seems like Scala 2.13 build fails as below:

```
[error] /home/runner/work/spark/spark/sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java:263:1:  error: cannot find symbol
[error]       System.exit(SparkExitCode.EXIT_SUCCESS);
[error]                                ^  symbol:   variable EXIT_SUCCESS
[error]   location: class SparkExitCode
[error] Note: Some input files use or override a deprecated API.
[error] Note: Recompile with -Xlint:deprecation for details.
[error] 1 error
```

**Comment by AngersZhuuuu** [comment]
> Tests are running in https://github.com/AngersZhuuuu/spark/runs/6765635006
> 
> Seems like Scala 2.13 build fails as below:
> 
> ```
> [error] /home/runner/work/spark/spark/sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java:263:1:  error: cannot find symbol
> [error]       System.exit(SparkExitCode.EXIT_SUCCESS);
> [error]                                ^  symbol:   variable EXIT_SUCCESS
> [error]   location: class SparkExitCode
> [error] Note: Some input files use or override a deprecated API.
> [error] Note: Recompile with -Xlint:deprecation for details.
> [error] 1 error
> ```

Updated

**Comment by AngersZhuuuu** [comment]
Looks like the latest failed test not related to this pr

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java (hunk 2 above)
Comment (cloud-fan) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java:263]: ah it's java...

**Comment by cloud-fan** [comment]
@AngersZhuuuu can you retrigger the tests?

**Comment by AngersZhuuuu** [comment]
> @AngersZhuuuu can you retrigger the tests?

GA passed now

**Pull request closed without merging**